pkg/virt-launcher/virtwrap/agent-poller/agent_poller.go

```diff
@@ -0,0 +1,236 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2018 Red Hat, Inc.
+ *
+ */
+package agentpoller
+
+import (
+	"encoding/json"
+	"fmt"
+	"reflect"
+	"time"
+
+	"k8s.io/apimachinery/pkg/types"
+
+	"kubevirt.io/kubevirt/pkg/log"
+	"kubevirt.io/kubevirt/pkg/virt-launcher/virtwrap/api"
+	"kubevirt.io/kubevirt/pkg/virt-launcher/virtwrap/cli"
+)
+
+type AgentUpdateEvent struct {
+	InterfaceStatuses *[]api.InterfaceStatus
+	DomainName        string
+}
+
+type AgentPoller struct {
+	Connection      cli.Connection
+	VmiUID          types.UID
+	domainData      DomainData
+	agentDone       chan struct{}
+	domainUpdate    chan *api.Domain
+	pollTime        time.Duration
+	agentUpdateChan chan AgentUpdateEvent
+}
+
+type DomainData struct {
+	name       string
+	aliasByMac map[string]string
+	interfaces []api.InterfaceStatus
+}
+
+// Result for json unmarshalling
+type Result struct {
+	Interfaces []Interface `json:"return"`
+}
+
+// Interface for json unmarshalling
+type Interface struct {
+	MAC  string `json:"hardware-address"`
+	IPs  []IP   `json:"ip-addresses"`
+	Name string `json:"name"`
+}
+
+// IP for json unmarshalling
+type IP struct {
+	IP     string `json:"ip-address"`
+	Type   string `json:"ip-address-type"`
+	Prefix int    `json:"prefix"`
+}
+
+type InterfaceData struct {
```

Comment:
Where do you use this one?

Reply:
Done
(Leftover from recent refactoring)

Reply:
createDomainData - this is actually needed, we look at updated domain and create domain data used later while processing agent updates